test/tests/functional/pbs_hook_alarm_large_multinode_job.py

```diff
@@ -0,0 +1,224 @@
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestPbsHookAlarmLargeMultinodeJob(TestFunctional):
+
+    """
+    This test suite contains hook test to verify that a large
+    nulti-node job does not slow down hook execution to cause an alarm.
+    """
+
+    def setUp(self):
+
+        TestFunctional.setUp(self)
+
+        a = {'resources_available.mem': '1gb',
+             'resources_available.ncpus': '1'}
+        self.server.create_vnodes(self.mom.shortname, a, 5000, self.mom)
+
+    def test_begin_hook(self):
+        """
+        Create an execjob_begin hook, import a hook content with a small
+        alarm value, and test it test it against a large multi-node job.
+        """
+        hook_name = "testhook"
+        hook_event = "execjob_begin"
+        hook_body = """
+import pbs
+e=pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG, "executing begin hook %s" % (e.hook_name))
+"""
+        a = {'event': hook_event, 'enabled': 'True',
+             'alarm': '15'}
+        self.server.create_import_hook(hook_name, a, hook_body)
+
+        j = Job(TEST_USER)
+        a = {'Resource_List.select': '5000:ncpus=1:mem=1gb',
+             'Resource_List.walltime': 10}
+
+        j.set_attributes(a)
+        j.set_sleep_time(10)
+
+        try:
+            jid = self.server.submit(j)
+        except PbsSubmitError:
+            pass
+
+        d = self.server.expect(JOB, {'job_state': 'R'},
+                               jid, max_attempts=15, interval=2)
+        self.assertTrue(d)
+
+        self.mom.log_match(
+            "pbs_python;executing begin hook %s" % (hook_name,), n=100,
+            max_attempts=5, interval=5, regexp=True)
+
+        self.mom.log_match(
+            "Job;%s;alarm call while running %s hook" % (jid, hook_event),
+            n=100, max_attempts=5, interval=5, regexp=True, existence=False)
+
+        self.mom.log_match("Job;%s;Started, pid" % (jid,), n=100,
+                           max_attempts=5, interval=5, regexp=True)
+
+    def test_prolo_hook(self):
+        """
+        Create an execjob_prologue hook, import a hook content with a
+        small alarm value, and test it test it against a large multi-node job.
+        """
+        hook_name = "testhook"
+        hook_event = "execjob_prologue"
+        hook_body = """
+import pbs
+e=pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG, "executing prologue hook %s" % (e.hook_name))
```

Comment:
This is fine with me.  Just keep in mind your tests differ from all other PTL tests.  If you are fine with this, then keep it as a tuple.